src/main/java/hudson/remoting/Engine.java

```diff
@@ -1038,11 +1057,14 @@ private static FileInputStream getFileInputStream(final File file) throws Privil
 
     @CheckForNull
     @Restricted(NoExternalUse.class)
-    static SSLSocketFactory getSSLSocketFactory(List<X509Certificate> x509Certificates)
+    static SSLContext getSSLContext(List<X509Certificate> x509Certificates, boolean noCertificateCheck)
             throws PrivilegedActionException, KeyStoreException, NoSuchProviderException, CertificateException,
             NoSuchAlgorithmException, IOException, KeyManagementException {
-        SSLSocketFactory sslSocketFactory = null;
-        if (x509Certificates != null && !x509Certificates.isEmpty()) {
+        SSLContext sslContext = null;
+        if (noCertificateCheck) {
+            sslContext = SSLContext.getInstance("TLS");
+            sslContext.init(null, new TrustManager[]{new NoCheckTrustManager()}, new SecureRandom());
+        } else if (x509Certificates != null && !x509Certificates.isEmpty()) {
```

Comment:
Moved from `JnlpAgentEndpointResolver#openURLConnection` to facilitate code reuse.